packages/node/src/indexer/dictionary.service.ts

```diff
@@ -232,4 +238,61 @@ export class DictionaryService implements OnApplicationShutdown {
     }
     return buildQuery(vars, nodes);
   }
+
+  async getSpecVersion(): Promise<SpecVersion[]> {
+    const { query } = this.specVersionQuery();
+    try {
+      const resp = await this.client.query({
+        query: gql(query),
+      });
+      const specVersionBlockHeightSet = new Set<SpecVersion>();
+      const _metadata = resp.data._metadata;
+      const specVersions = resp.data.specVersions.nodes;
+
+      if (specVersions && specVersions.length >= 0) {
+        // Add range for the last specVersion
+        if (_metadata.lastProcessedHeight) {
+          specVersionBlockHeightSet.add({
+            id: specVersions[specVersions.length - 1].id,
+            start: Number(specVersions[specVersions.length - 1].blockHeight),
+            end: Number(_metadata.lastProcessedHeight),
+          });
+        }
+        // Add range for -1 specVersions
+        for (let i = 0; i < resp.data.specVersions.nodes.length - 1; i++) {
+          specVersionBlockHeightSet.add({
+            id: specVersions[i].id,
+            start: Number(specVersions[i].blockHeight),
+            end: Number(specVersions[i + 1].blockHeight) - 1,
+          });
+        }
+      }
+      return Array.from(specVersionBlockHeightSet);
+    } catch (err) {
+      logger.warn(err, `failed to fetch specVersion result`);
+      return undefined;
+    }
+  }
+
+  private specVersionQuery(): GqlQuery {
```

Comment:
Do we need to worry about the fact that we limit queries to 100 results?

Reply:
For now I think we are safe, might remove this query limit just for specVersion in query service.